documentation.md

```diff
@@ -0,0 +1,97 @@
+# Turbo for Django
+
+This repository aims to provide utilities for working with [Turbo](https://turbo.hotwire.dev)
+with the Django web framework. 
+
+## Turbo Streams
+Currently, the repository contains utilities for working with
+[Turbo Streams](https://turbo.hotwire.dev/handbook/streams) over Websockets, the one part of
+Turbo which requires a specific integration with the backend framework to function. In Django's
+case, this means taking advantage of [Django Channels](https://github.com/django/channels).
+
+The Django Streams integration is heavily inspired by the helpers and mixins found in
+[turbo-rails](https://github.com/hotwired/turbo-rails). There are a few steps required to get
+the integration working.
+
+### Initial Setup
+#### Django Channels
+After setting up Channels according to the [documentation](https://channels.readthedocs.io/en/stable/installation.html),
+make sure to modify the top-level `ProtocolTypeRouter` to include the `TurboStreamsConsumer`:
+
+```python
+from turbo.consumers import TurboStreamsConsumer
+application = ProtocolTypeRouter({
+  "http": AsgiHandler(),
+  "websocket": TurboStreamsConsumer.as_asgi()  # Leave off .as_asgi() if using Channels 2.x
```

Comment:
Are we planning to support Channels < 3? Would be nice to document support expectations near the top, Django and Channels versions.

Reply:
Everything should work with Channels 2.x as long as you leave off the `as_asgi()` method call, since 2.x consumers are asgi apps in and of themselves